### DISCUSSION TITLE
Keycloack 502 Bad Request

### DISCUSSION THREAD

**Discussion by ahmad-septian** [keycloak/keycloak#24789] [Q&A]
### Description

Has anyone ever used Keycloak? 

I have a problem with my Keycloak version 22, so this is when I login to my project with Keycloak it's safe to enter my project application but when I try to reload automatically 502 bad request why is that? after I traced it I checked the network the results when I first logged in there was a token in the network but when I reloaded the web the token disappeared on the network like that do you have a solution?
I have tried to set the token as in this code that I will provide but still can't, what I expect is that when the Web Reload is not 502 Bad Request

is this an error from the coding or from the Keycloak side?

<img width="1440" alt="image" src="https://github.com/keycloak/keycloak/assets/93763528/39036e2c-3744-463e-b838-7dd68242e58d">

**Comment by schuerg** [comment]
It could be caused by too big HTTP headers sent from your browser to your loadbalancer or reverse proxy. I faced this in combination with keycloak and too big access tokens resulting in too big Authentication bearer http headers. The loadbalancer/reverse proxy then answers with an http 502 error. Keycloak never receives the http request in this scenario.

The solution is to either reduce the number of oidc claims in your tokens or to increase the allowed http header size in the loadbalancer/reverse proxy.

**Comment by Qmeriaux** [comment]
Did you manage to make it work ? I have the same issue

**Comment by China-Chris** [comment]
@stianst I also encountered this problem, and the reason is that the token is too long. I changed the Default Signature Algorithm to ES384 and solved this problem
